internal/provider/common/vm/disk.go

```diff
@@ -248,25 +249,20 @@ DiskCreate
 
 creates a detachable disk.
 */
-func DiskCreate(ctx context.Context, inVDC vdc.VDC, vm *govcd.VM, disk *Disk, inVapp vapp.VApp) (*Disk, diag.Diagnostics) {
+func DiskCreate(ctx context.Context, org org.Org, vdc vdc.VDC, vm *govcd.VM, disk *Disk, inVapp vapp.VAPP) (*Disk, diag.Diagnostics) {
```

Comment:
why we use vdc vdc.VDC ?
This information is already given in struct vapp.VAPP, isn't it ?

Reply:
VDC in VAPP is not exported and this is the desired operation